### PR TITLE
[windows] Adds support for keyboard mapping.

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

Adds support for Windows keyboard, adding RawKeyEventDataWindows. The embedder counter part is a WIP PR.

## Related Issues

https://github.com/flutter/flutter/issues/37710

## Tests

Updated `raw_keyboard_test.dart` and the event simulator.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
Looks great!

32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: dev/tools/gen_keycodes/lib/dart_code_gen.dart [hunk 1]
```diff
@@ -225,6 +225,43 @@ $otherComments  static const LogicalKeyboardKey $constantName = LogicalKeyboardK
...
+      if (entry.windowsKeyCodes != null){
+        for (final int code in entry.windowsKeyCodes) {
+          windowsNumPadMap.writeln('  $code: LogicalKeyboardKey.${entry.constantName},');
+        }
+    }
+    }
+    return windowsNumPadMap.toString().trimRight();
+  }
+
+  /// This generates the map of Android key codes to logical keys.
```
Comment (stuartmorgan) [inline dev/tools/gen_keycodes/lib/dart_code_gen.dart:252]:
```suggestion
  /// This generates the map of Windows key codes to logical keys.
```

File: packages/flutter/lib/src/services/raw_keyboard_windows.dart [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+import 'package:flutter/foundation.dart';
+
+import 'keyboard_key.dart';
+import 'keyboard_maps.dart';
+import 'raw_keyboard.dart';
+
+/// Platform-specific key event data for Windows.
+///
+/// This object contains information about key events obtained from Windows's
+/// `NSEvent` interface.
```
Comment (stuartmorgan) [inline packages/flutter/lib/src/services/raw_keyboard_windows.dart:14]: Seems unlikely that it's from `NSEvent` ;)

File: packages/flutter/lib/src/services/raw_keyboard_windows.dart [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        return modifiers & leftMask != 0 && modifiers & rightMask != 0 || anyOnly;
+      case KeyboardSide.left:
+        return modifiers & leftMask != 0 || anyOnly;
+      case KeyboardSide.right:
+        return modifiers & rightMask != 0 || anyOnly;
+    }
+    return false;
+  }
+
+    @override
```
Comment (stuartmorgan) [inline packages/flutter/lib/src/services/raw_keyboard_windows.dart:124]:
```suggestion
  @override
```

**Review by stuartmorgan-g — APPROVED** [review]
Just tiny nits. Thanks for tackling this!

File: packages/flutter/lib/src/services/raw_keyboard_windows.dart (hunk 2 above)
Comment (franciscojma86) [reply packages/flutter/lib/src/services/raw_keyboard_windows.dart]: 🤦‍♂

**Comment by franciscojma86** [comment]
@googlebot I fixed it.

**Comment by yjbanov** [comment]
This added 4KB of gzipped size to the "hello world" app when compiled to the web. Is it possible to use `import ... as deferred` so we don't eagerly load all the data? /cc @ferhatb who is looking at code size

Screen Shot 2020-04-08 at 1 30 09 PM

**Comment by ferhatb** [comment]
@yjbanov , the string based switch makes it untreeshakable. For web i filed an issue long ago to treeshake keymaps out. However, this must be adding to ios/android code size as well, is there a size_compile_test for those that shows regression as well ?

**Pull request merged**